src/Spd.Resource.Repository/SharedContactFuncs.cs

```diff
@@ -172,7 +172,7 @@ private static spd_address CreateAddrFromContact(contact c, bool isMailing)
 
     private static contact UpdateExistingContact(contact existingContact, contact newContact)
     {
-        if (newContact.address1_country != null && newContact.address1_line1 != null && newContact.address1_city != null)
+        if (!string.IsNullOrEmpty(newContact.address1_country) && !string.IsNullOrEmpty(newContact.address1_line1) && !string.IsNullOrEmpty(newContact.address1_city))
```

Comment:
if address is empty, this cannot clear the data. 
That's not what we want.

Reply:
I reverted the changes for mappings and update func